src/libraries/System.Collections.Immutable/tests/Frozen/FrozenSetTests.cs

```diff
@@ -270,5 +270,66 @@ protected override ICollection NonGenericICollectionFactory(int count)
         protected override Type ICollection_NonGeneric_CopyTo_ArrayOfIncorrectValueType_ThrowType => typeof(InvalidCastException);
 
         protected override Type ICollection_NonGeneric_CopyTo_NonZeroLowerBound_ThrowType => typeof(ArgumentOutOfRangeException);
+
+        [Fact]
+        public void Sparse_LookupItems_AlltemsFoundAsExpected()
+        {
+            foreach (int size in new[] { 1, 2, 10, 63, 64, 65, 999, 1024 })
+            {
+                foreach (int skip in new[] { 2, 3, 5 })
+                {
+                    var original = new HashSet<int>(Enumerable.Range(-3, size).Where(i => i % skip == 0));
+                    var frozen = original.ToFrozenSet();
+
+                    for (int i = -10; i <= size + 66; i++)
+                    {
+                        Assert.Equal(original.Contains(i), frozen.Contains(i));
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void ClosedRange_Lookup_AllItemsFoundAsExpected()
+        {
+            foreach (long start in new long[]
+                {
+                    long.MinValue,
+                    (long)int.MinValue - 1,
+                    int.MinValue,
+                    -1,
+                    0,
+                    1,
+                    int.MaxValue - 1,
+                    int.MaxValue,
+                    (long)int.MaxValue + 1,
+                    uint.MaxValue - 1,
+                    uint.MaxValue,
+                    (long)uint.MaxValue + 1,
+                    long.MaxValue - 5
+                })
+            {
+                foreach (long size in new[] { 1, 2, 5 })
+                {
+                    var original = new HashSet<long>();
+
+                    var min = start;
+                    var max = start + size - 1;
+                    for (long i = min; i != max; i++)
+                    {
+                        original.Add(i);
+                    }
+
+                    var frozen = original.ToFrozenSet();
+
+                    min = start > long.MinValue ? start - 10 : start;
+                    max = start + size - 1 < long.MaxValue ? start + size + 9 : start + size - 1;
+                    for (long i = min; i != max; i++)
+                    {
+                        Assert.Equal(original.Contains(i), frozen.Contains(i));
+                    }
+                }
+            }
+        }
```

Comment:
There's a lot of src code being added and not a lot of tests.  Code coverage looks ok, for both existing code (where the additions might shift some existing tests to use the new implementations) and new code?

Reply:
Turns out the new implementations "stole" some of the existing test cases in a natural way. I augmented where some new edges needed to be tested.

Is there specific places where you'd like more coverage?

Reply:
Added tests to tickle the edges of all the new types.